### ISSUE TITLE
[$1000] Auto-completed number & dropdown-arrow aren't visible in Phone Input of Onfido

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#14879]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Open the app
2. Go to any workspace
3. Connect bank account,
4. Fill all the information till Personal information
5. Select any country for "Issuing Country" & Go to next step
6. Click on "Get secure link"
7. Click on "Get link via SMS"
8. Notice that the dropdown isn't visible for country selector
9. if You already have entered your number before then use auto-complete number and you'll notice that it's not visible

## Expected Result:
Number & country selector arrow should be visible

## Actual Result:
Auto-completed number & country selector arrow aren't visible

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?

- [ ] MacOS / Chrome / Safari
- 

**Version Number:** 1.2.65-2
**Reproducible in staging?:** y
**Reproducible in production?:** could not check
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/217077926-895bee66-534f-49bf-b307-049332e65a5c.mp4

https://user-images.githubusercontent.com/43996225/217077961-0f7d6b01-f422-4a2e-8bf7-6814b06a4085.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @daraksha-dk 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1675612872131759

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01f262576653c70610</li>
        <li>Upwork Job ID: 1623780597168963584</li>
        <li>Last Price Increase: 2023-02-09</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @isabelastisser (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by isabelastisser** [comment]
I will review by EOD.

**Comment by isabelastisser** [comment]
I was able to reproduce this, adding the external label.

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01f262576653c70610

**Comment by MelvinBot** [comment]
Current assignee @isabelastisser is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @mananjadhav (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @alex-mechler (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The country flag image and dropdown arrow does not show. Also, when we do an autofill, the text is not visible.

### What is the root cause of that problem?
1. The country flag image does not show is related to the Content Security Policy. You will find this console error when trying to load the image. It does not happen on dev.

![image](https://user-images.githubusercontent.com/50919443/218021075-2ceeb76d-0ce0-4e5e-9c45-3743cd205703.png)

2. The arrow dropdown is not visible because it takes the color from `colorContentBody` we set at onfido `customUI`.
3. The autofill text has a white color
https://github.com/Expensify/App/blob/14ebe26b6af9c19b7debedb25bcbdf976cb57c5e/web/index.html#L59-L71

### What changes do you think we should make in order to solve the problem?


1. I think the flag issue should be handled internally.

2 & 3.
In my opinion, we don't need to change the css specifically for the phone number input. If we look at the issuing country input, the input has a transparent background and a text color based on `colorContentBody`. 

![image](https://user-images.githubusercontent.com/50919443/218020095-73ba585c-0c1f-4ba3-853b-38e2102620f9.png)

But that's not the case with phone number input. So, I think the input style is inconsistent and I already open an issue upstream [here](https://github.com/onfido/onfido-sdk-ui/issues/2144). After the style is fixed, it will look like this

![image](https://user-images.githubusercontent.com/50919443/218022346-561fdf64-5cc3-4643-a82f-f33201305a1f.png)

_the right border width has always been 0. There is currently no way to customize it without changing it directly from the css_

### What alternative solutions did you explore? (Optional)
If the phone number input is intentionally designed with white background and the onfido team is not going to "fix" it, then we should use @daraksha-dk proposal.

**Comment by mananjadhav** [comment]
Thanks for the formatted proposals folks. I agree with @bernhardoj's analysis and I think this should be fixed on onfido sdk side.

@alex-mechler Do you think we can connect with onFido team from Expensify side?

**Comment by alex-mechler** [comment]
> @alex-mechler Do you think we can connect with onFido team from Expensify side?

I've set myself a reminder to check on the issue @bernhardoj created (thanks) later in the week to see if they have replied there. If they have not I'll reach out to them more directly. Given that this is something that _can_ be fixed by overwriting the css, I imagine this will be low priority if they are willing to fix it. They have shown in the past they will place things with CSS overrides as fairly low priority.

**Comment by alex-mechler** [comment]
Reached out to Onfido to see if they can prioritize this. Going to put a HOLD in the title since this is held on them at this point

**Comment by alex-mechler** [comment]
Onfido replied, they currently have a backlog of issues, but an engineer will be assigned asap

**Comment by isabelastisser** [comment]
@alex-mechler, since we are waiting on Onfido, should we move this to weekly?

**Comment by alex-mechler** [comment]
Good call, bumping it down to a weekly!

**Comment by alex-mechler** [comment]
Update from Onfido 
<img width="1291" alt="image" src="https://user-images.githubusercontent.com/22447860/220685904-9dd52a70-f190-43b9-9ca3-ff2c2f129f6b.png">

**Comment by isabelastisser** [comment]
Not overdue, waiting for Onfido.

**Comment by alex-mechler** [comment]
Update from Onfido: 
<img width="1255" alt="image" src="https://user-images.githubusercontent.com/22447860/226387618-75f4c6cb-b828-411d-af0b-34ecb9026634.png">

**Comment by mananjadhav** [comment]
I am not sure what would be the next steps here. Will it be holding the issue indefinitely till we get a fix from Onfido? or we just close this one out?

**Comment by isabelastisser** [comment]
Not overdue, on hold!

**Comment by isabelastisser** [comment]
Not overdue, on hold!

**Comment by michaelhaxhiu** [comment]
Removing AlexM's assignment and assigning a new CME.

**Comment by Julesssss** [comment]
I emailed Onfido to see if there is a more recent update.

**Comment by Julesssss** [comment]
<img width="872" alt="Screenshot 2023-06-26 at 08 32 57" src="https://github.com/Expensify/App/assets/10736861/0f12a1f8-8d02-4489-b328-72a71e161ce0">

They responded, but we're already on the 'fixed' version. We updated the lib 2 months ago, so perhaps this is the case and the bug no longer exists? 

```
"onfido-sdk-ui": "12.2.1",
"@onfido/react-native-sdk": "7.4.0",
```

**Comment by bernhardoj** [comment]
Looks like it's not fully fixed.

Only the number input background becomes dark (not even transparent)
<img width="219" alt="image" src="https://github.com/Expensify/App/assets/50919443/f8bc518e-e15b-4f2c-9c2e-50bca5ca2668">

Expected:
<img width="357" alt="image" src="https://github.com/Expensify/App/assets/50919443/211780d9-a02e-4c30-b22f-7167352d7641">

**Comment by mananjadhav** [comment]
@Julesssss I was wondering if there's a CSS fix we should do for this one? If we're open to that then we can take this off hold?

**Comment by Julesssss** [comment]
I emailed Onfido support already so let's see what they say first (hi, if you're reading this 👋)

**Comment by mananjadhav** [comment]
Got it. Thanks @Julesssss

**Comment by mananjadhav** [comment]
@Julesssss did we hear anything from Onfido?

**Comment by Julesssss** [comment]
Not yet, I just asked for an update

**Comment by Julesssss** [comment]
Fix should be about 2 weeks away!

<img width="878" alt="Screenshot 2023-07-14 at 12 25 26" src="https://github.com/Expensify/App/assets/10736861/2de45f0a-c0dd-4196-afde-9bb6be74d2c3">

**Comment by mananjadhav** [comment]
Thanks for the update @Julesssss. Will we receive an update when the fix is pushed?

**Comment by Julesssss** [comment]
I'm not sure, but worst case we'll just have to keep checking :)

**Comment by isabelastisser** [comment]
Still on hold.

**Comment by isabelastisser** [comment]
Still on hold.

**Comment by isabelastisser** [comment]
Still on hold!

**Comment by isabelastisser** [comment]
Still on hold.

**Comment by Julesssss** [comment]
Ah, they never informed me. But by now they suraly have released the fix... 

Lets unhold, retest to see if it still occurs, and if not we can try to bump the library.

**Comment by Julesssss** [comment]
Excellent. I'm not sure what payment would be due here though. It was created with the old payment rules

**Comment by mananjadhav** [comment]
I think I am eligible for the partial payout because I spent time in reviewing the proposals earlier.

**Comment by Julesssss** [comment]
Agree that you and @bernhardoj spent time triaging this and verifying a few of the onfido library updates... 

Perhaps $250 each? Does that sound reasonable?

**Comment by mananjadhav** [comment]
> Agree that you and @bernhardoj spent time triaging this and verifying a few of the onfido library updates...

Yes.

> Perhaps $250 each? Does that sound reasonable?

Yes works for me.

**Comment by Julesssss** [comment]
Great, thanks.

@isabelastisser could you please pay out? This has been resolved for a while now so there's no need to HOLD payment.

**Comment by isabelastisser** [comment]
Payment summary:

- C+ review $250 @mananjadhav (manual request in NewDot)
- @bernhardoj $250 proposal (payment in Upwork)
- @daraksha-dk $250 reported issue (payment in Upwork)

**Comment by isabelastisser** [comment]
All set!

**Comment by JmillsExpensify** [comment]
$250 payment approved for @mananjadhav based on comment above.
